src/backend/distributed/commands/role.c

```diff
@@ -1241,12 +1241,6 @@ PreprocessGrantRoleStmt(Node *node, const char *queryString,
 		return NIL;
 	}
 
-	/*
-	 * Postgres don't seem to use the grantor. Even dropping the grantor doesn't
-	 * seem to affect the membership. If this changes, we might need to add grantors
-	 * to the dependency resolution too. For now we just don't propagate it.
-	 */
```

Comment:
LGTM now except what this comment states ..

As I mentioned earlier, could you please talk to @halilozanakgul to decide:
a) How hard it is to accomplish what this comment states, or,
b) If we're taking the risk of "<grantor> role does not exist" errors coming from workers when the grantor is not a distributed role given that we expect such conditions to be rare.

Thanks!

Reply:
nvm this, but let's fix metadata syncing issue in this PR too

Reply:
Fixed the grantor propagation other than missing grantor before metadata sync phase. In dependency resolution phase, if grantor has admin rolesi then it can not be propagated. I think it is not a must for our tasks. I can open an issue and continue working on that task with suitable priority